hotness/consumers.py

```diff
@@ -796,140 +733,230 @@ def get_dist_tag(self):
+
+        bz = old_task[0]
+        upstream = new_task[2]
+
+        if old_task[1] == 'CLOSED':
+            tmp = tempfile.mkdtemp(prefix='thn-rh-', dir='/var/tmp')
+            try:
+                rh_stuff = None
+                try:
+                    result_rh, rh_stuff = self.buildsys.check_rebase(
+                        upstream, old_task_id, new_task_id, tmp)
+                    if new_task[1] == 'CLOSED':
+                        self.bugzilla.follow_up("Scratch builds compared successfully.", bz)
+                    elif new_task[1] == 'FAILED':
+                        self.bugzilla.follow_up("Scratch build of rebased version failed.", bz)
+                    elif new_task[1] == 'CANCELED':
+                        self.bugzilla.follow_up("Scratch build of rebased version "
+                                                "has been canceled.", bz)
+                    self._rh_attach_logs(bz, rh_stuff)
+                except Exception as e:
+                    self.log.warning("rebase-helper failed with: %s" % str(e))
+                    self.log.warning(traceback.format_exc())
+                    self._rh_fail(e, bz, rh_stuff)
+
+            finally:
+                self.log.debug("Removing %r" % tmp)
+                shutil.rmtree(tmp)
+
+        elif old_task[1] == 'FAILED':
+            self.bugzilla.follow_up("Scratch build of rawhide version failed.", bz)
+        elif old_task[1] == 'CANCELED':
+            self.bugzilla.follow_up("Scratch build of rawhide version has been canceled.", bz)
+
+    def _register_tasks(self, package, version, upstream, bz, rh_stuff):
+        """
+        Add tasks initiated by rebase-helper to tracking dicts.
+
+        Task ids are retrieved from rh_stuff dictionary containing data
+        from rebase-helper.
+
+        Returns ids of both tasks.
+        """
+        old_task_id = None
+        new_task_id = None
+
+        for ver in ['old', 'new']:
+            try:
+                task_id = rh_stuff['build_logs']['build_ref'][ver]['koji_task_id']
+                if ver == 'old':
+                    self.old_triggered_task_ids[task_id] = [
+                        bz, None, str(version), str(package)]
+                    old_task_id = task_id
+                else:
+                    self.new_triggered_task_ids[task_id] = [
+                        bz, None, str(upstream), str(package)]
+                    new_task_id = task_id
+            except KeyError:
+                pass
+
+        return old_task_id, new_task_id
+
+    def _find_tasks(self, task_id):
+        """
+        Find pair of tasks sharing the same bug id by task id.
+
+        Returns pair of task ids.
+        """
+        old_task_id = None
+        new_task_id = None
+
+        # Find tasks with common bug id
+        if task_id in self.old_triggered_task_ids:
+            old_task_id, old_task = task_id, self.old_triggered_task_ids[task_id]
```

Comment:
Not that it'd save much, but you could probably define this as a function (maybe just internal to this one) since it's two identical blocks. It's not a big deal either way to me, so just pick whatever you like.